integration/integration_tms_upload_test.go

```diff
@@ -0,0 +1,123 @@
+	}
+	container.assertHasOutput(t, "description: Piper integration test")
+	container.assertHasOutput(t, "tmsUpload - File uploaded successfully")
+	container.assertHasOutput(t, "tmsUpload - Node upload executed successfully")
+	container.assertHasOutput(t, "tmsUpload - SUCCESS")
+}
+
+func TestTmsUploadIntegrationBinNoDescriptionSuccess(t *testing.T) {
+	// success case: run cmd without --nodeExtDescriptorMapping and --customDescription
+	readEnv()
+	container := givenThisContainer(t, IntegrationTestDockerExecRunnerBundle{
+		Image:       "devxci/mbtci-java11-node14",
+		User:        "root",
+		TestDir:     []string{"testdata", "TestTmsUploadIntegration/bin_param"},
+		Environment: map[string]string{"PIPER_tmsServiceKey": tmsServiceKey},
+	})
+	defer container.terminate(t)
+
+	err := container.whenRunningPiperCommand("tmsUpload",
+		"--mtaPath=scv_x.mtar",
+		"--nodeName=PIPER-TEST",
+		"--mtaVersion=1.0.0",
+		"-v")
+	if err != nil {
+		t.Fatalf("Piper command failed %s", err)
+	}
+	container.assertHasOutput(t, "description: Created by Piper")
+	container.assertHasOutput(t, "tmsUpload - File uploaded successfully")
+	container.assertHasOutput(t, "tmsUpload - Node upload executed successfully")
+	container.assertHasOutput(t, "tmsUpload - SUCCESS")
+}
+
+func TestTmsUploadIntegrationBinFailParam(t *testing.T) {
+	// run cmd with nodeExtDescriptorMapping
+	readEnv()
+	container := givenThisContainer(t, IntegrationTestDockerExecRunnerBundle{
+		Image:   "devxci/mbtci-java11-node14",
+		User:    "root",
+		TestDir: []string{"testdata", "TestTmsUploadIntegration/bin_param"},
+	})
+	defer container.terminate(t)
+
+	err := container.whenRunningPiperCommand("tmsUpload",
+		"--mtaPath=scv_x.mtar",
+		"--nodeName=PIPER-TEST",
+		"--customDescription=Piper integration test",
+		"--nodeExtDescriptorMapping={\"PIPER-TEST\":\"scv_x.mtaext\", \"PIPER-PROD\":\"scv_x.mtaext\"}",
+		"--mtaVersion=1.0.0",
+		"-v")
+
+	assert.Error(t, err, "Did expect error")
+	container.assertHasOutput(t, "Error: unknown flag: --nodeExtDescriptorMapping")
+}
+
+func TestTmsUploadIntegrationYaml(t *testing.T) {
+	// success case: run with config.yaml
+	readEnv()
+	container := givenThisContainer(t, IntegrationTestDockerExecRunnerBundle{
+		Image:       "devxci/mbtci-java11-node14",
+		User:        "root",
+		TestDir:     []string{"testdata", "TestTmsUploadIntegration/yaml"},
+		Environment: map[string]string{"PIPER_tmsServiceKey": tmsServiceKey},
+	})
+	defer container.terminate(t)
+
+	err := container.whenRunningPiperCommand("tmsUpload")
+	if err != nil {
+		t.Fatalf("Piper command failed %s", err)
+	}
+
+	container.assertHasOutput(t, "tmsUpload - File uploaded successfully")
+	container.assertHasOutput(t, "tmsUpload - MTA extension descriptor updated successfully")
+	container.assertHasOutput(t, "tmsUpload - Node upload executed successfully")
+	container.assertHasOutput(t, "tmsUpload - SUCCESS")
+	//  test that oauth token is not exposed
+	container.assertHasNoOutput(t, "eyJ")
```

Comment:
Ideally, that would be good to test that we also do not print client secret into the logs (both in usual and in base64 encoded format), but I do not know, how to do it in a meaningful way. Probably, that would be enough to regisiter it as a secret for the log and rely on this functionality. See e.g. https://github.com/SAP/jenkins-library/blob/master/pkg/tms/tms.go#L116